### DISCUSSION TITLE
스크린샷 모듈을 가져와서 사용해보는데 스샷이 본래 화면과 다르게 찍힙니다

### DISCUSSION THREAD

**Discussion by cwcc-cjlim** [naverz/zepeto-studio-kor#634] [5. Q&A]
![KakaoTalk_20220419_090728562](https://user-images.githubusercontent.com/103082113/163895797-518dbadf-0f14-4281-83c2-67d9d8cccda5.jpg)
![KakaoTalk_20220419_090728562_01](https://user-images.githubusercontent.com/103082113/163895801-499e8c24-a206-40a9-b077-4c7ff706121e.png)

예제에 있는 카메라 모듈을 패키지 형태로 가져와서 썼는데 어느 부분을 살펴 봐야할지 몰라서 스크린샷만 우선 올려봅니다 ㅜㅜ
(+포스트 프로세싱을 빼니 스크린샷 결과물이 정상적으로 나옵니다)

**Comment by Zepeto-TechEvangelist** [comment]
확인해 보겠습니다.

**Reply by cwcc-cjlim** [reply]
부탁드리겠습니다!

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 확인 결과  'Post-process Layer'의 Anti-aliasing 모드가 FXAA 또는 TAA 로 적용되어 있을 경우 발생하는 것으로 보입니다.
유니티 자체 문제로 판단되어 이슈 해결에는 시간이 걸릴 것 같아서 우선 우회적인 해결안 안내 드립니다.

기본 모드를 No Anti-aliasing 으로 하거나, 추가적인 코드를 넣어서 사진 찍는 동안에만 No Anti-aliasing로 설정해 보세요. 

![image](https://user-images.githubusercontent.com/100118650/164615999-b09cac91-83b6-4d4a-be21-4a1a4838ad0c.png)

ScreenShotController.ts 에 아래 코드를 추가해 보세요.

<hr/>

private preAntialiasingMode:Antialiasing;   
    *RenderTargetTextureWithBackground()
    {
        yield new WaitForEndOfFrame();
        this.camera.Render();
        this.camera.targetTexture = null;

        // 코드 추가 
        this.camera.GetComponent<PostProcessLayer>().antialiasingMode = this.preAntialiasingMode;
    }

    private TakeScreenShotWithBackground() {
        // 타겟 텍스처를 지정하고 카메라 렌더
        this.camera.targetTexture = this.renderTexture;

        // 코드 추가 
        this.preAntialiasingMode = this.camera.GetComponent<PostProcessLayer>().antialiasingMode;
        this.camera.GetComponent<PostProcessLayer>().antialiasingMode = Antialiasing.None;
    
        this.StartCoroutine(this.RenderTargetTextureWithBackground());
    }

**Reply by cwcc-cjlim** [reply]
감사합니다!

**Comment by minjujuu** [comment]
- [[포스트프로세싱 사용하기]](https://github.com/naverz/zepeto-studio-kor/discussions/451) 
위 링크에 작성했던 방식으로 Selfie Camera 프리팹이랑 Zepeto Camera 프리팹에 적용해보았을 땐 발생하지 않았는데, 
혹시 어떤 방식으로 포스트 프로세싱을 적용하셨을까요!?
- 모바일 앱에서 저장하기 눌러서 실제 저장 시에도 두 번째 사진처럼 사진이 저장되나요? 
- 포스트 프로세싱 볼륨에서 이런 저런 효과를 추가해봐도 발생하지 않았는데, 혹시 어떤 효과를 적용하셨는지 알 수 있을까요?
![image](https://user-images.githubusercontent.com/79794414/164454549-55d95994-bb91-4933-a2bf-37396628e81e.png)

추가로 아래처럼 텍스처에 투명한 부분이 있는 나무 같은 경우 
뒷 배경이 흰색이면 말씀주신 사진처럼 비슷하게 보이는 것 같은데, 혹시 사진 뒤에 깔린 이미지가 흰색이 있지는 않으신가용?
![image](https://user-images.githubusercontent.com/79794414/164455152-019c737a-ff54-4c6e-9106-a878ce6b4290.png)

**Reply by cwcc-cjlim** [reply]
뒤에 배경 관계없이 본문 사진처럼 찍히게 됩니다 ㅜㅜ 
포스트프로세싱 적용한 카메라와 스크린샷모듈을 가져올때 당시 패키지 목록을 같이 올려보겠습니다
![카메라01](https://user-images.githubusercontent.com/103082113/164563446-5de9d8d8-64a1-4b05-be96-c3ad372e6ab2.PNG)
![카메라02](https://user-images.githubusercontent.com/103082113/164563451-2bf5d6eb-51df-4157-b150-681c515f3197.PNG)
![카메라03](https://user-images.githubusercontent.com/103082113/164563452-a4c0d098-0acf-4ddd-9993-d89f2ae4f22e.PNG)
![카메라04](https://user-images.githubusercontent.com/103082113/164563453-05fbb3d5-186d-44cb-befa-77678ece9e8d.PNG)
![카메라05](https://user-images.githubusercontent.com/103082113/164563455-fb3ae16b-73f9-400e-ab3b-4d1d13190e74.PNG)
